Libraries/LibGfx/Bitmap.cpp

```diff
@@ -160,6 +160,10 @@ void Bitmap::apply_mask(Gfx::Bitmap const& mask, MaskKind mask_kind)
 
 ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::cropped(Gfx::IntRect crop, Gfx::Color outside_color) const
 {
+    // OPTIMIZATION: Skip slow manual copying for NO-OP crops
+    if (crop == rect())
+        return clone();
```

Comment:
While this makes sense, I suspect this method is mainly slow because of the format limitation you removed in the previous commit: we needed to have pixel-to-pixel conversion happening.

Now you have an output bitmap that always has the same format as the input bitmap, you should be able to fill a bitmap with the outside color (if larger than `rect()`) and do `TypedTransfer<T>::copy` operations on partial or whole scanlines instead.